pkg/events/events_test.go

```diff
@@ -68,3 +71,119 @@ func TestEmptyFields(t *testing.T) {
 	_, err = createEventRecord(si.EventRecord_QUEUE, "obj", "group", "", "message")
 	assert.Assert(t, err != nil, "the EventRecord should not be created with empty reason")
 }
+
+func TestEmitReserveEventWithoutEventCache(t *testing.T) {
+	cache := GetEventCache()
+	assert.Assert(t, cache == nil, "cache should not be initialized")
+
+	allocKey := "test-alloc-1"
+	appID := "app-1"
+	nodeID := "node-1"
+	err := EmitReserveEvent(allocKey, appID, nodeID)
+	assert.NilError(t, err, "emitting event without event cache should succeed")
+}
+
+func TestEmitReserveEvent(t *testing.T) {
+	CreateAndSetEventCache()
+	defer ResetCache()
+	cache := GetEventCache()
+	cache.StartService()
+
+	allocKey := "test-alloc-2"
+	appID := "app-2"
+	nodeID := "node-2"
+	err := EmitReserveEvent(allocKey, appID, nodeID)
+	assert.NilError(t, err, "expected EmitReserveEvent to run without errors")
+
+	assertEmitRequestAppAndNodeEvents(t, allocKey, appID, nodeID, "AppReservedNode")
```

Comment:
Test coverage for the error cases could be easily extended by passing in empty strings.
That would cover the codecov remarks

Reply:
Added tests for those in the `events` package, but I could not get triggered the errors in `scheduling_application.go`.